cloudflare/resource_cloudflare_custom_hostname_fallback_origin.go

```diff
@@ -0,0 +1,123 @@
+package cloudflare
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/pkg/errors"
+)
+
+func resourceCloudflareCustomHostnameFallbackOrigin() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudflareCustomHostnameFallbackOriginCreate,
+		Read:   resourceCloudflareCustomHostnameFallbackOriginRead,
+		Update: resourceCloudflareCustomHostnameFallbackOriginUpdate,
+		Delete: resourceCloudflareCustomHostnameFallbackOriginDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudflareCustomHostnameFallbackOriginImport,
+		},
+
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"zone_id": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"origin": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceCloudflareCustomHostnameFallbackOriginRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	zoneID := d.Get("zone_id").(string)
+
+	customHostnameFallbackOrigin, err := client.CustomHostnameFallbackOrigin(zoneID)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("error reading custom hostname fallback origin %q", zoneID))
+	}
+
+	d.Set("origin", customHostnameFallbackOrigin.Origin)
+	d.Set("status", customHostnameFallbackOrigin.Status)
+
+	return nil
+}
+
+func resourceCloudflareCustomHostnameFallbackOriginDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	zoneID := d.Get("zone_id").(string)
+
+	err := client.DeleteCustomHostnameFallbackOrigin(zoneID)
+	if err != nil {
+		return errors.Wrap(err, "failed to delete custom hostname fallback origin")
+	}
+
+	return nil
+}
+
+func resourceCloudflareCustomHostnameFallbackOriginCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	zoneID := d.Get("zone_id").(string)
+	origin := d.Get("origin").(string)
+
+	fallbackOrigin := cloudflare.CustomHostnameFallbackOrigin{
+		Origin: origin,
+	}
+
+	_, err := client.UpdateCustomHostnameFallbackOrigin(zoneID, fallbackOrigin)
+	if err != nil {
+		return errors.Wrap(err, "failed to create custom hostname fallback origin")
+	}
+
+	d.SetId(zoneID + "-custom_hostname_fallback_origin")
```

Comment:
Can we pass this through to [`stringChecksum`](https://github.com/cloudflare/terraform-provider-cloudflare/blob/f28c2bb96e78247690f690e65b9ebe4589ccd275/cloudflare/utils.go#L60-L66) to maintain consistent ID formats? It will return an MD5 hash and match the other identifiers we have in the state.

Reply:
Done.